### PR TITLE
Milestone-Three-Feature-Branch - Make BooksForm and RemoveBook Functions to work properly.

### PR DESCRIPTION
**Pull request by happiguru**
This Pull request implements the following functionalities:

1. Adds `handleTitleChange`, `handleCategoryChange` and `handleSubmit` functions to the BooksForm component.
2. Add newly created books to the Redux Store.
3. Add another `removeBook` function that deletes a book from the store.

Thank you.

### CODE REVIEW

File: src/reducers/index.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
-import { combineReducers } from 'redux';
+import { combineReducers, createStore } from 'redux';
 import booksReducer from './books';
 
-export default combineReducers({ books: booksReducer });
+const store = createStore(
+  combineReducers({ books: booksReducer }),
+);
```
Comment (bellom) [inline src/reducers/index.js]: I think your store should be created in `src/index.js` from milestone 2 requirement.

File: src/index.js [hunk 2]
```diff
@@ -2,12 +2,16 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
 import { Provider } from 'react-redux';
-import { createStore } from 'redux';
+// import { createStore } from 'redux';
```
Comment (bellom) [inline src/index.js]: Always remove commented code, it is better to use git history.

File: src/index.js [hunk 3]
```diff
@@ -2,12 +2,16 @@ import React from 'react';
...
-import books from './reducers/books';
+import { createBook } from './actions/index';
+// import books from './reducers/books';
 import App from './components/App';
+import store from './reducers/index';
 
-const store = createStore(books, initialState);
+// const store = createStore(books, initialState);
+
+initialState.map((book) => store.dispatch(createBook(book)));
```
Comment (bellom) [inline src/index.js]: Oops 🤦 , Let's try to stick to best practice, you made this correction in milestone 2 to my surprise you changed it back, are you having any difficulty doing it the right way?

Kindly update this, 👍 


image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @happiguru  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check my comments under this review;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

File: src/reducers/index.js (hunk 1 above)
Comment (happiguru) [reply src/reducers/index.js]: Fixed this issue.

File: src/index.js (hunk 2 above)
Comment (happiguru) [reply src/index.js]: Fixed the issue.

File: src/index.js (hunk 3 above)
Comment (happiguru) [reply src/index.js]: Please I seem not to be able to get around this. Your help will be much appreciated.

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hello @happiguru**

Great job with the project so far it looks amazing, but there are still some changes to be made:
 
---
### Requested 🔴 

- [x] Regarding the way you are defining the initial state. the reason that the method the previous TSE suggested doesn't work with you is the way you wrote the initial state. 
The initial state is supposed to be an object that contains a key called books and the value of that key is the array of books you have. Take a look at this picture for reference.


Screenshot 2021-08-03 171255

---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

File: src/index.js (hunk 3 above)
Comment (bellom) [reply src/index.js]: @happiguru So sorry I'm just seeing this, I was out.

I hope it is all good now.

Comment (happiguru) [reply src/index.js]: Yes it is okay now. Thank you!!

**Review by bellom — APPROVED** [review]
## Hi @happiguru  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Pull request merged**